llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -790,6 +800,32 @@ bool CallLowering::handleAssignments(ValueHandler &Handler,
       CCValAssign &VA = ArgLocs[j + Idx];
       const ISD::ArgFlagsTy Flags = Args[i].Flags[Part];
 
+      // We found an indirect parameter passing and we are at the first part of
+      // the value being passed. In this case copy the incoming pointer into a
+      // virtual register so later we can load it.
+      if (VA.getLocInfo() == CCValAssign::Indirect && Flags.isSplit()) {
+        if (Handler.isIncomingArgumentHandler()) {
+          Handler.assignValueToReg(ArgReg, VA.getLocReg(), VA);
+          Handler.assignValueToAddress(Args[i].OrigRegs[Part],
+                                       Args[i].Regs[Part], OrigTy,
+                                       MachinePointerInfo{}, VA);
+        } else {
+          MachineFrameInfo &MFI = MF.getFrameInfo();
+          int FrameIdx = MFI.CreateStackObject(OrigTy.getScalarSizeInBits(),
+                                               Align(8), false);
```

Comment:
Maybe `TargetFrameLowering`'s `getStackAlign` would do the job here?

If it would, the I will have to check it whether I have access to a `TargetFrameLowering` in this callback.

Reply:
SelectionDAG on RISC-V uses

```
      Align StackAlign =                                                         
          std::max(getPrefTypeAlign(Outs[OutIdx].ArgVT, DAG),                    
                   getPrefTypeAlign(ArgValue.getValueType(), DAG));
```

Reply:
```MF.getSubtarget().getFrameLowering()```